### ISSUE TITLE
Speed improvements for API?

### ISSUE THREAD

**Issue by brandomr** [BeckySharp/OdinsonWebapp#6]
@BeckySharp I'm curious if you have any ideas for how we could improve the speed/performance of the `/process_text` API?

It takes ~30+ seconds to process each article on the fly. If there are any low hanging fruit for improving the speed that would be very helpful on our end.

**Comment by BeckySharp** [comment]
probably -- can you send me a couple of articles and I can profile it?

**Comment by brandomr** [comment]
will do

**Comment by BeckySharp** [comment]
TL;DR  Overall the main way to speed it up, apparently, is to give it more memory (!)

ok, so I profiled the running and found a couple of things:
- the speed is GREATLY improved with more memory. According to flight recorder, the system is happy with about 6-7G of memory, so if you can spare it, I'd def recommend giving it more like 6 or 8G.  With more memory it was taking about 10sec per document.
- most of the time is spent annotating the text for syntax.  To address this, I modified the annotation to be as minimal as I feel is reasonable, but you'll prob only see gains on larger documents.  

That said -- though we're currently using Processors as the annotation library, and here in the lab we're pretty happy with it, Odinson itself is library agnostic.  So, if you find a faster parser you'd like to use, you/we can work on a pipeline that incorporates that instead.

There were also 3 docs that failed, Marco and I debugged this and fixed it in Odinson, so getting a fresh pull and publishing local again (or re-building your docker container) will fix those. Any future barfs would be really cool to know about so we can continue to improve things!

I won't close this, in case you try memory and you're still having performance issues, but if you find it helps, you can close it.  If not, I'll make a second optimization effort.

👍

**Comment by brandomr** [comment]
Just ran 236 docs and it took ~4 seconds per document 👍

**Comment by BeckySharp** [comment]
yay!

**Comment by BeckySharp** [comment]
I went ahead and modified the `.sbtopts` file on master to have 8g
